tests/test_elffile.py

```diff
@@ -0,0 +1,100 @@
+import io
+import pathlib
+import struct
+
+import pytest
+
+from packaging._elffile import EIClass, EIData, ElfFile, ElfInvalid, EMachine
+
+DIR_MANYLINUX = pathlib.Path(__file__, "..", "manylinux").resolve()
+DIR_MUSLLINUX = pathlib.Path(__file__, "..", "musllinux").resolve()
+BIN_MUSL_X86_64 = DIR_MUSLLINUX.joinpath("musl-x86_64").read_bytes()
+
+
+@pytest.mark.parametrize(
+    "name, capacity, encoding, machine",
+    [
+        ("x86_64-x32", EIClass.C32, EIData.Lsb, EMachine.X8664),
+        ("x86_64-i386", EIClass.C32, EIData.Lsb, EMachine.I386),
+        ("x86_64-amd64", EIClass.C64, EIData.Lsb, EMachine.X8664),
+        ("armv7l-armel", EIClass.C32, EIData.Lsb, EMachine.Arm),
+        ("armv7l-armhf", EIClass.C32, EIData.Lsb, EMachine.Arm),
+        ("s390x-s390x", EIClass.C64, EIData.Msb, EMachine.S390),
+    ],
+)
+def test_elffile_glibc(name, capacity, encoding, machine):
+    path = DIR_MANYLINUX.joinpath(f"hello-world-{name}")
+    with path.open("rb") as f:
+        ef = ElfFile(f)
+        assert ef.capacity == capacity
+        assert ef.encoding == encoding
+        assert ef.machine == machine
+        assert ef.flags is not None
+
+
+@pytest.mark.parametrize(
+    "name, capacity, encoding, machine, interpreter",
+    [
+        (
+            "aarch64",
+            EIClass.C64,
+            EIData.Lsb,
+            EMachine.AArc64,
+            "aarch64",
+        ),
+        ("i386", EIClass.C32, EIData.Lsb, EMachine.I386, "i386"),
+        ("x86_64", EIClass.C64, EIData.Lsb, EMachine.X8664, "x86_64"),
+    ],
+)
+def test_elffile_musl(name, capacity, encoding, machine, interpreter):
+    path = DIR_MUSLLINUX.joinpath(f"musl-{name}")
+    with path.open("rb") as f:
+        ef = ElfFile(f)
+        assert ef.capacity == capacity
+        assert ef.encoding == encoding
+        assert ef.machine == machine
+        assert ef.interpreter == f"/lib/ld-musl-{interpreter}.so.1"
+
+
+@pytest.mark.parametrize(
+    "data",
+    [
+        # Too short for magic.
+        b"\0",
+        # Enough for magic, but not ELF.
+        b"#!/bin/bash" + b"\0" * 16,
+        # ELF, but unknown byte declaration.
+        b"\x7fELF\3" + b"\0" * 16,
+    ],
+    ids=["no-magic", "wrong-magic", "unknown-format"],
```

Comment:
And now I know `ids` exists! 😅

Reply:
There’s also `pytest.param(..., id=...)`

Reply:
Just learned about that on Twitter when I tweeted about this! 😅